src/WrappedMToken.sol

```diff
@@ -301,15 +340,15 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
         _revertIfInsufficientAmount(amount_);
         _revertIfInvalidRecipient(recipient_);
 
-        if (_accounts[recipient_].isEarning) {
+        if (_accounts[recipient_].earningState == EarningState.NOT_EARNING) {
```

Comment:
introduce _isEarning to not flip logic like here

Reply:
Slightly more gas, bnt I guess the tradeoff for less changes (no logic flip) is worth it. I'll do it.